pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -266,12 +266,12 @@ def on_train_start(self, trainer, pl_module):
         trainer.ckpt_path = ckpt_path
         trainer.weights_save_path = ckpt_path
 
-    @rank_zero_only
     def on_validation_end(self, trainer, pl_module):
-        # only run on main process
-        if trainer.global_rank != 0:
-            return
-
+        # To get checkpointing working on TPU, need to call _save_model
+        # for all ranks, to avoid deadlocks.  Assuming save_function is mapped
```

Comment:
nit: without `if trainer.global_rank != 0:` and `@rank_zero_only`, all threads are writing to the log